security/medusa/l3/registry.c

```diff
@@ -147,26 +91,43 @@ struct medusa_kclass_s * med_get_kclass_by_pointer(struct medusa_kclass_s * ptr)
  * med_unregister_kclass soon.
  */
 
-int med_unlink_kclass(struct medusa_kclass_s * ptr)
+int _med_unlink_kclass(struct medusa_kclass_s *med_kclass)
 {
-	int retval = 0;
-	struct medusa_kclass_s * tmp;
+	struct medusa_kclass_s *pos, *prev;
+
+	if (med_kclass == kclasses) {
+		kclasses = med_kclass->next;
+		return 0;
+	}
+
+	for (prev = kclasses, pos = kclasses->next; pos != NULL; prev = pos, pos = pos->next) {
+		if (pos == med_kclass) {
+			prev->next = pos->next;
+			return 0;
+		}
+	}
+
+	med_pr_warn("Failed to unlink kclass '%s' - not found", med_kclass->name);
```

Comment:
See comments on med_unlink_kclass(): "If the use-count is nonzero, it fails gracefully. This allows use of med_unlink_kclass as an atomic uninstallation check & unlink. Always check the return value of this call."

med_pr_warn is not so gracefully ;)    ---> remove med_pr_warn(), or put it between #ifdef DEBUG ... #endif /* DEBUG */

Reply:
I changed it as med_pr_devel, since pr_devel only prints messages #if DEBUG